src/PluginRTCPeerConnectionConstraints.swift

```diff
@@ -4,49 +4,76 @@ import Foundation
 class PluginRTCPeerConnectionConstraints {
 	fileprivate var constraints: RTCMediaConstraints
 
+	// https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/createOffer#RTCOfferOptions_dictionary
+	// TODO voiceActivityDetection?
+	fileprivate static let allowedConstraints : Array = ["iceRestart", "OfferToReceiveVideo", "OfferToReceiveAudio"]
 
 	init(pcConstraints: NSDictionary?) {
 		NSLog("PluginRTCPeerConnectionConstraints#init()")
-
-		if pcConstraints == nil {
-			self.constraints = RTCMediaConstraints()
-			return
-		}
-
-		var	offerToReceiveAudio = pcConstraints?.object(forKey: "offerToReceiveAudio") as? Bool
-		var	offerToReceiveVideo = pcConstraints?.object(forKey: "offerToReceiveVideo") as? Bool
-
-		if offerToReceiveAudio == nil && offerToReceiveVideo == nil {
-			self.constraints = RTCMediaConstraints()
-			return
+		var mandatoryConstraints: [String : String] = [:]
+		var optionalConstraints: [String : String] = [:]
+		
+		let _mandatoryConstraints = pcConstraints?.object(forKey: "mandatory") as? NSDictionary
+		let _optionalConstraints = pcConstraints?.object(forKey: "optional") as? NSDictionary
```

Comment:
I believe optional constraints will be in an array like, `[{ googDscp: true }, { DtlsSrtpKeyAgreement: true }]` so this will need to be `let _optionalConstraints = pcConstraints?.object(forKey: "optional") as? NSArray` and handled differently in `parseConstraints`

Reply:
@cah-dunn Thank you for review.
For now we only support the following constraints when mandatory and optional is an object. mandatory and optional are out of spec since 2014 BTW.

> Array = ["iceRestart", "OfferToReceiveVideo", "OfferToReceiveAudio"]
see:  https://github.com/cordova-rtc/cordova-plugin-iosrtc/blob/task/libwebrtc-update-m69/src/PluginRTCPeerConnectionConstraints.swift#L9

DtlsSrtpKeyAgreement is also depcreated.
- https://bugs.chromium.org/p/chromium/issues/detail?id=804275

This are Chrome "custom proprietary constrains in libwebrtc" I will see if I implement them tomorrow.
- https://github.com/versatica/mediasoup-client/issues/38#issuecomment-445770418

Reply:
@cah-dunn See PR https://github.com/cordova-rtc/cordova-plugin-iosrtc/pull/406
If you can help testing would be great.

Reply:
@hthetiot Awesome, thank you. I'll test these changes. I'm guessing I could probably remove these constraints from our project given how long ago they were added (we use the same code for our iPad app as our web app w/ Chrome, etc).